Tests/OpenAPIGeneratorReferenceTests/SnippetBasedReferenceTests.swift

```diff
@@ -2524,18 +2536,36 @@ final class SnippetBasedReferenceTests: XCTestCase {
                         public var single: Swift.String?
                         public var manyExploded: [Swift.String]?
                         public var manyUnexploded: [Swift.String]?
+                        public struct sortPayload: Codable, Hashable, Sendable {
+                            public var option1: Swift.String?
+                            public var option2: Swift.String?
+                            public init(
+                                option1: Swift.String? = nil,
+                                option2: Swift.String? = nil
+                            ) {
+                                self.option1 = option1
+                                self.option2 = option2
+                            }
+                            public enum CodingKeys: String, CodingKey {
+                                case option1
+                                case option2
+                            }
+                        }
+                        public var sort: Operations.get_sol_foo.Input.Query.sortPayload
                         public init(
                             single: Swift.String? = nil,
                             manyExploded: [Swift.String]? = nil,
-                            manyUnexploded: [Swift.String]? = nil
+                            manyUnexploded: [Swift.String]? = nil,
+                            sort: Operations.get_sol_foo.Input.Query.sortPayload
```

Comment:
```suggestion
                            sort: Operations.get_sol_foo.Input.Query.sortPayload = nil
```

This should end up being defaulted to nil as well. It's possible it'll require improvements in inspecting the nested object and whether it can be initialized with `.init()`.

Reply:
That parameter has `required: true` though, I don't think it should be defaulted to`nil`. It should be defaulted to `.init()` instead I think?

Reply:
Oh that's interesting! What does it mean for a query parameter of type object to be required, when all properties are optional?

On the wire, it won't look any different to the whole parameter being optional.

Maybe we should just change the sample to have an optional parameter? I think that's more common. Alternatively, make one of the properties required. Either is fine.

Reply:
Right, that's interesting. Updating the sample to have an optional parameter would resolve the issues you highlighted in the review with this test. 

What if all the properties of the object have default values though? I tried getting an idea of what it would look like by updating the tests but I can get any default value to show in the generated swift types.